index.ts

```diff
@@ -196,9 +206,13 @@ async function init() {
 		type: 'blob',
 	});
 
-	const filename
-		= query.get('filename')
-		?? `${user} ${repository} ${gitReference} ${directory}`.replace(/\//, '-');
+	const filename = query.get('filename')
+		?? [
+			user,
+			repository,
+			gitReference,
+			directory ? directory.replaceAll('/', '-') : undefined,
+		].filter(Boolean).join('-');
```

Comment:
The tool should not be used to download URLs like `https://github.com/user/repo`. Rather, it should already detect them and point you to the native archive download (much faster). 

If you want to add support private repos in that case, it should be added by fetching the archive link and downloading _that one_ (which I assume already happens on public repos but not on private ones?)